lib/chainlock/chainlock.js

```diff
@@ -22,310 +22,310 @@ const bls = require('../crypto/bls');
  * @property {Buffer} blockHash
  * @property {Buffer} signature
  */
-function ChainLock(arg) {
-  if (arg instanceof ChainLock) {
-    return arg.copy();
+class ChainLock {
+  constructor(arg) {
+    if (arg instanceof ChainLock) {
+      return arg.copy();
+    }
+    this.toJSON = this.toObject;
```

Comment:
Looks hacky. I would just call `toObject` from `toJSON` method.

Reply:
We usually use `ChainLock.prototype.toJSON = ChainLock.prototype.toObject;` type of format for this, and it is actually nice to inherits JSDoc for IDE (which can still be fixed with documentation ts file).

Reply:
Is it ok for both of you now?

Reply:
Hm... Not sure that we usually doing like this. Could you provide examples, please? I wasn’t able to find anything like this in repos where we don’t have any legacy and strctly follow our code style (DPP, dapi-client, drive)